vllm/engine/llm_engine.py

```diff
@@ -655,6 +655,7 @@ def _add_processed_request(
         lora_request: Optional[LoRARequest],
         prompt_adapter_request: Optional[PromptAdapterRequest],
         trace_headers: Optional[Mapping[str, str]] = None,
+        priority: Optional[int] = None,
     ) -> None:
         self._validate_model_inputs(processed_inputs)
```

Comment:
We may want to check here if priority is provided (or nonzero per other comment), that the scheduler is in priority mode, and if not raise an exception. Not 100% sure about this though.